commons/src/test/java/io/aiven/kafka/tieredstorage/commons/SegmentCompressionCheckerTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.FileRecords;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MemoryRecordsBuilder;
+import org.apache.kafka.common.record.TimestampType;
+
+import org.assertj.core.util.Files;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SegmentCompressionCheckerTest {
+
+    @TempDir
+    Path dir;
+
+    @Test
+    void shouldRequireCompression_whenCompressionNone() throws IOException {
```

Comment:
Should we have 2 parameterized tests one for case when compression changes and one when it does not?

Reply:
Not sure the complexity of the tests justify parametrized tests at this point. IIUC, It would require passing compression type + expected output as parameters which adds overhead when reading the tests. 
Though, if I'm missing some scenarios we could consider it.

Reply:
On a second thought, I agree parametrized tests could help here. I gave a try, lmk how it looks.

Reply:
LGTM